### PR TITLE
fix recursive call to _substmerge (issue #60)

### PR DESCRIPTION
**Pull request by zerwes**
issue #60

### CODE REVIEW

File: hiyapyco/__init__.py [hunk 1]
```diff
@@ -368,10 +368,10 @@ def _substmerge(self, a, b):
                         logger.debug('deepmerge dict: set key %s' % k)
                         a[k] = b[k]
             elif isinstance(b, listTypes):
-                logger.debug('deepmerge: dict <- list ... "%s" <- "%s"' % (a, b,))
+                logger.debug('substmerge: dict <- list ... "%s" <- "%s"' % (a, b,))
                 for bd in b:
                     if isinstance(bd, dict):
-                        a = self._deepmerge(a, bd)
+                        a = self._substmerge(a, bd)
```
Comment (reddart) [inline hiyapyco/__init__.py:374]: @zerwes The Same change is also needed in line 366 and that should be it.

**Review by reddart — APPROVED** [review]
Thanks a lot @zerwes for taking the time. Looks good to me. Tested with my use cases.

**Comment by zerwes** [comment]
Hello again @reddart
Well, I hope I got it all sorted.
As the PR #37 came without some test I had implemented some after the merge.
But as the PR included some errors, the PR #38 trying to fix these, resulted in failing tests.
I inspected the failing tests, and indeed they where wrong in terms of the consequent implementation of _substmerge_.
IMHO now all should work as expected.
Please have a look at the changes in `hiyapyco/__init__.py`and `test/test_merge.py`.
If you are fine with them, the merge can be completed.
Thank you for bringing this up again and helping clearing it.

**Comment by zerwes** [comment]
Thank you @reddart ; a new release will be out soon

**Pull request merged**